### DISCUSSION TITLE
WhatsApp

### DISCUSSION THREAD

**Discussion by GarryLowther** [buildnatively/community#83] [Ideas]
### Use case

```Text
Users wish to send messages via WhatsApp from phone numbers inside our app.
```

### Idea

```Text
Add native WhatsApp to Natively to allow:
1. Send message to phone number
2. Read all phone numbers and other details
3. Add contacts to WhatsApp
```

**Comment by romanfurman6** [comment]
@GarryLowther Hey, I think you're able to do that right now. @vladdehtiarov @victormagdalyuk

**Comment by victormagdalyuk** [comment]
@GarryLowther As of today, there is no official public WhatsApp SDK due to restrictions placed by WhatsApp. They don't want developers to integrate WhatsApp functionality into apps.

1. Send message to phone number.
Partially possible: An app can utilize a WhatsApp Click to Chat link. This link, when clicked by the user, opens a chat window with a pre-populated phone number in WhatsApp. However, the app itself cannot directly initiate messages within WhatsApp due to privacy restrictions.

2. Read all phone numbers and other details.
Not possible: WhatsApp prioritizes user privacy and doesn't allow apps to access user contact lists or phone numbers directly. This functionality would violate WhatsApp's terms of service.

3. Add contacts to WhatsApp.
Not possible: Similar to reading contact details, adding contacts directly to WhatsApp from an app is not allowed due to privacy restrictions.

A reference link: https://www.whatsapp.com/legal/terms-of-service#terms-of-service-acceptable-use-of-our-services:~:text=create%20software%20or%20APIs%20that%20function%20substantially%20the%20same%20as%20our%20Services%20and%20offer%20them%20for%20use%20by%20third%20parties%20in%20an%20unauthorized%20manner
